sklearn/utils/validation.py

```diff
@@ -777,6 +777,13 @@ def check_array(
         if all(isinstance(dtype_iter, np.dtype) for dtype_iter in dtypes_orig):
             dtype_orig = np.result_type(*dtypes_orig)
 
+    elif hasattr(array, "iloc") and hasattr(array, "dtype"):
```

Comment:
For my education: why not use `isinstance()` instead of "guessing" that this is a `pd.Series` based on the presence of attributes?

Reply:
`isinstance(pd.Series)` requires to import pandas which may not be installed since it's not a dependency of scikit-learn

Reply:
Makes sense, there are a few parts (outside of examples and benchmarks) where the `try: import pandas ... except ImportError: pass` pattern is used (for example in `_param_validation.py`). Maybe worth pondering some kind of consolidation, but for a new issue. Or maybe people already pondered it in the past.

Reply:
For the examples and benchmark, we are happy to depend on pandas (it is indeed documented). The idea is to not depend on pandas in the code itself.

In the parameter validation, we use `pd` because this is the only way to handle the `pd.NA` beast. We, therefore, try to import it and if pandas is not installed, we know that `pd.NA` will not be available in anyway.